module/executiondatasync/provider/provider_test.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/onflow/flow-go/module/executiondatasync/provider"
+	"github.com/onflow/flow-go/module/executiondatasync/tracker"
+	mocktracker "github.com/onflow/flow-go/module/executiondatasync/tracker/mock"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func getDatastore() datastore.Batching {
+	return dssync.MutexWrap(datastore.NewMapDatastore())
+}
+
+func getExecutionDataStore(ds datastore.Batching) execution_data.ExecutionDataStore {
+	return execution_data.NewExecutionDataStore(blobs.NewBlobstore(ds), execution_data.DefaultSerializer)
+}
+
+func getBlobservice(ds datastore.Batching) network.BlobService {
+	blobstore := blobs.NewBlobstore(ds)
+	blobService := new(mocknetwork.BlobService)
+	blobService.On("AddBlobs", mock.Anything, mock.AnythingOfType("[]blocks.Block")).Return(blobstore.PutMany)
+	return blobService
+}
+
+func getProvider(blobService network.BlobService) *provider.Provider {
+	trackerStorage := new(mocktracker.Storage)
+	trackerStorage.On("Update", mock.Anything).Return(func(fn tracker.UpdateFn) error {
+		return fn(func(uint64, ...cid.Cid) error { return nil })
+	})
+
+	return provider.NewProvider(
+		zerolog.Nop(),
+		metrics.NewNoopCollector(),
+		execution_data.DefaultSerializer,
+		blobService,
+		trackerStorage,
+	)
+}
+
+func generateChunkExecutionData(t *testing.T, minSerializedSize uint64) *execution_data.ChunkExecutionData {
+	ced := &execution_data.ChunkExecutionData{
+		TrieUpdate: &ledger.TrieUpdate{
+			Payloads: []*ledger.Payload{
+				{
+					Value: nil,
+				},
+			},
+		},
+	}
+
+	size := 1
+
+	for {
+		buf := &bytes.Buffer{}
+		require.NoError(t, execution_data.DefaultSerializer.Serialize(buf, ced))
+
+		if buf.Len() >= int(minSerializedSize) {
+			t.Logf("Chunk execution data size: %d", buf.Len())
+			return ced
+		}
+
+		v := make([]byte, size)
+		_, _ = rand.Read(v)
+		ced.TrieUpdate.Payloads[0].Value = v
+		size *= 2
+	}
+}
+
+func generateBlockExecutionData(t *testing.T, numChunks int, minSerializedSizePerChunk uint64) *execution_data.BlockExecutionData {
+	bed := &execution_data.BlockExecutionData{
+		BlockID:             unittest.IdentifierFixture(),
+		ChunkExecutionDatas: make([]*execution_data.ChunkExecutionData, numChunks),
+	}
+
+	for i := 0; i < numChunks; i++ {
+		bed.ChunkExecutionDatas[i] = generateChunkExecutionData(t, minSerializedSizePerChunk)
+	}
+
+	return bed
+}
+
+func TestHappyPath(t *testing.T) {
+	t.Parallel()
+
+	ds := getDatastore()
+	provider := getProvider(getBlobservice(ds))
+	store := getExecutionDataStore(ds)
+
+	test := func(numChunks int, minSerializedSizePerChunk uint64) {
+		expected := generateBlockExecutionData(t, numChunks, minSerializedSizePerChunk)
+		executionDataID, err := provider.Provide(context.Background(), 0, expected)
+		require.NoError(t, err)
+		actual, err := store.GetExecutionData(context.Background(), executionDataID)
```

Comment:
the test doesn't hang, it's just that the comparison `assert.DeepEquals` takes a very very long time.

Also, it doesn't return before blobs are actually stored, because we always wait for the channel to close, which happens after blobs are stored